### ISSUE TITLE
[ceph] Validate and update the existing plugin

### ISSUE THREAD

**Issue by badone** [sosreport/sos#496]
Check that the existing Ceph plugin is up to date and comprehensive enough for the current Ceph releases.

**Comment by bmr-cymru** [comment]
Thanks for working on this Brad - if there are changes the command/file sets for specific distros (e.g. RHEL, Debian) we should handle that using the tagging class mechanism. Right now we have a single ceph for all distros:

```python
class Ceph(Plugin, RedHatPlugin, UbuntuPlugin):
    """CEPH distributed storage
    """

    plugin_name = 'ceph'
    profiles = ('storage', 'virt')

    option_list = [
        ("log", "gathers all ceph logs", "slow", False)
    ]

    packages = (
        'ceph',
        'ceph-mds',
        'ceph-common',
        'libcephfs1',
        'ceph-fs-common'
    )

    def setup(self):
        self.add_copy_spec([
            "/etc/ceph/",
            "/var/log/ceph/"
        ])

        self.add_cmd_output([
            "ceph status",
            "ceph health",
            "ceph osd tree",
            "ceph osd stat",
            "ceph osd dump",
            "ceph mon stat",
            "ceph mon dump"
        ])

        self.add_forbidden_path("/etc/ceph/*keyring")
        self.add_forbidden_path("/var/lib/ceph/*/*keyring")
        self.add_forbidden_path("/var/lib/ceph/*keyring")
```

If there's a need to add or remove files or commands we can do something like:

```python
class Ceph(Plugin, UbuntuPlugin):
        # [...]

class RedHatCeph(Ceph, RedHatPlugin):
    def setup(self):
        super(RedHatCeph, self).setup()
        self.add_copy_spec("/etc/special-redhat-ceph-stuff")
```

Anything that's cross-distro should stay in the base Ceph class though.

**Comment by badone** [comment]
After review and discussions with stakeholders ended up with https://github.com/sosreport/sos/pull/521

**Comment by badone** [comment]
Here is the patch, sorry this took so long and hope here is okay.

diff --git a/sos/plugins/ceph.py b/sos/plugins/ceph.py
index 52b8fff..651abba 100644
--- a/sos/plugins/ceph.py
+++ b/sos/plugins/ceph.py
@@ -42,12 +42,14 @@ class Ceph(Plugin, RedHatPlugin, UbuntuPlugin):
 
         self.add_cmd_output([
             "ceph status",
-            "ceph health",
+            "ceph health detail",
             "ceph osd tree",
             "ceph osd stat",
             "ceph osd dump",
             "ceph mon stat",
-            "ceph mon dump"
+            "ceph mon dump",
+            "ceph df",
+            "ceph report"
         ])
 
         self.add_forbidden_path("/etc/ceph/*keyring")

**Comment by pmoravec** [comment]
As there were unresolved issues in PR #558 , I created #591.

@badone could you please review if the patch contains all requested data (was created based on rhbz#1210527)? Thanks.

**Comment by badone** [comment]
See my  note in #591

**Comment by bmr-cymru** [comment]
Merged:

```
commit 696f3c605c5d6b0cb8c0703635401991bc7af934
Author: Pavel Moravec <pmoravec@redhat.com>
Date:   Thu Jul 2 13:10:31 2015 +0200

    [ceph] add calamari-server and librados2 to the package list
    
    Enable ceph plugin also by either calamari-server or librados2
    package.
    
    This ensures Calamari and Rados log collection is effective
    regardless of which packages are installed on a given system.
    
    This is a temporary solution until rados is split from the ceph
    package.
    
    Resolves: #496
    
    Signed-off-by: Pavel Moravec <pmoravec@redhat.com>
    Signed-off-by: Bryn M. Reeves <bmr@redhat.com>

commit 24ecc933d109c0eba120b434d08aa560e9a56869
Author: Pavel Moravec <pmoravec@redhat.com>
Date:   Wed Jul 1 12:15:41 2015 +0200

    [ceph] add logs, report and calamari conf
    
    Update ceph commands and add log and configuration file
    collection for calamari and radosgw.
    
    Resolves: #496
    Resolves: rhbz#1210527
    
    Signed-off-by: Brad Hubbard <bhubbard@redhat.com>
    Signed-off-by: Pavel Moravec <pmoravec@redhat.com>
    Signed-off-by: Bryn M. Reeves <bmr@redhat.com>

```
